assembly.py

```diff
@@ -471,6 +472,52 @@ def order_and_orient(inFasta, inReference, outFasta,
     #    os.unlink(trimmed)
     return 0
 
+def _call_order_and_orient_orig(kwargs):
+    _order_and_orient_orig(**kwargs)
+
+def order_and_orient(inFasta, inReference, outFasta,
+        outAlternateContigs=None, outReference=None,
+        breaklen=None, # aligner='nucmer', circular=False, trimmed_contigs=None,
+        maxgap=200, minmatch=10, mincluster=None,
+        min_pct_id=0.6, min_contig_len=200, min_pct_contig_aligned=0.3, n_genome_segments=0, threads=0):
+    ''' This step cleans up the de novo assembly with a known reference genome.
+        Uses MUMmer (nucmer or promer) to create a reference-based consensus
+        sequence of aligned contigs (with runs of N's in between the de novo
+        contigs).
+    '''
+
+    ref_segments_all = tuple(Bio.SeqIO.parse(inReference, 'fasta'))
+    n_genome_segments = n_genome_segments or len(ref_segments_all)
+    assert (len(ref_segments_all) % n_genome_segments) == 0
+    n_refs = int(len(ref_segments_all) / n_genome_segments)
+    log.info('n_genome_segments={} n_refs={}'.format(n_genome_segments, n_refs))
+
+    with util.file.tempfnames([ '.{}.ref.fasta'.format(ref_num) for ref_num in range(n_refs)]) as refs_fasta, \
+         util.file.tempfnames([ '.{}.scaffold.fasta'.format(ref_num) for ref_num in range(n_refs)]) as scaffolds_fasta, \
+         util.file.tempfnames([ '.{}.altcontig.fasta'.format(ref_num) for ref_num in range(n_refs)]) as alt_contigs_fasta:
+
+         for ref_num in range(n_refs):
+             Bio.SeqIO.write(ref_segments_all[ref_num*n_genome_segments : (ref_num+1)*n_genome_segments], refs_fasta[ref_num], 'fasta')
+
+         pool = multiprocessing.Pool(threads or None)
+         pool.map(_call_order_and_orient_orig,
+                  [dict(inFasta=inFasta, inReference=refs_fasta[ref_num], outFasta=scaffolds_fasta[ref_num], 
```

Comment:
Prefer `{}` over `dict()` constructor.

Reply:
why?

Reply:
It's a google python style guide. I also see it in the wild much more than `dict()`

Reply:
If it's important to follow the Google python style guide I'll change it, but I prefer dict() for dictionaries that represent parameters of a function call, as opposed to more general dicts.  The syntax looks more like a function call, and permits adding **kwargs .